clients/src/main/java/org/apache/kafka/clients/consumer/internals/metrics/KafkaConsumerMetrics.java

```diff
@@ -91,16 +92,20 @@ public KafkaConsumerMetrics(Metrics metrics, String metricGrpPrefix) {
     }
 
     public void recordPollStart(long pollStartMs) {
+        if (this.pollEndMs != 0) {
+            long pollIntervalMs = pollStartMs - this.pollStartMs;
+            long pollTimeMs = this.pollEndMs - this.pollStartMs;
+            double pollIdleRatio = pollTimeMs * 1.0 / pollIntervalMs;
+            this.pollIdleSensor.record(pollIdleRatio);
+        }
         this.pollStartMs = pollStartMs;
         this.timeSinceLastPollMs = lastPollMs != 0L ? pollStartMs - lastPollMs : 0;
         this.timeBetweenPollSensor.record(timeSinceLastPollMs);
         this.lastPollMs = pollStartMs;
     }
 
     public void recordPollEnd(long pollEndMs) {
-        long pollTimeMs = pollEndMs - pollStartMs;
-        double pollIdleRatio = pollTimeMs * 1.0 / (pollTimeMs + timeSinceLastPollMs);
-        this.pollIdleSensor.record(pollIdleRatio);
+        this.pollEndMs = pollEndMs;
```

Comment:
think this wouldn't record a metric when there's only one invocation of poll(). This is also evident from the failing `KafkaConsumerTest::testPollIdleRatio` test.

Reply:
You are right. This is on purpose. If you only have 1 invocation of `poll`, then it doesn't make sense to ask "What fraction of the polling interval time was spent inside poll". The interval is not defined yet.

Thanks for letting me know about the test, I've adjusted the test to account for the changes. I also added a bit to demonstrate the issue: Repeatedly running poll while spending 10ms inside and outside the poll causes the metric to approach 0.33.

Please let me know if the changes look reasonable. I note that there's a similar metric in `KafkaShareConsumerMetricsTest`. Should that be updated too? If so, let me know and I'll apply the same changes to that one.